system/src/system_network_compat.cpp

```diff
@@ -255,6 +257,90 @@ class NetworkAccessTechnologyDiagnosticData: public AbstractIntegerDiagnosticDat
     }
 };
 
+class NetworkCellularCellGlobalIdentityMobileCountryCodeDiagnosticData
+    : public AbstractIntegerDiagnosticData
+{
+public:
+    NetworkCellularCellGlobalIdentityMobileCountryCodeDiagnosticData()
+        : AbstractIntegerDiagnosticData(
+              DIAG_ID_NETWORK_CELLULAR_CELL_GLOBAL_IDENTITY_MOBILE_COUNTRY_CODE,
+              DIAG_NAME_NETWORK_CELLULAR_CELL_GLOBAL_IDENTITY_MOBILE_COUNTRY_CODE)
+    {
+    }
+
+    virtual int get(IntType& val)
+    {
+        CellularGlobalIdentity cgi;
+        cellular_global_identity(&cgi, nullptr);
+        val = static_cast<IntType>(cgi.mobile_country_code);
+
+        return SYSTEM_ERROR_NONE;
+    }
+} g_networkCellularCellGlobalIdentityMobileCountryCodeDiagnosticData;
```

Comment:
Actually, since there are so few others (4, maybe 5), I think it may be better to change the rest of the file to follow this.

It keeps each message as a single block and allows for them to be organized much better. For instance, the block makes it easier to wrap such messages in a platform flag (i.e. `HAL_PLATFORM_CELLULAR`).

Reply:
No strong preference, but again I personally find it's easy to miss the instantiation :man_shrugging:

Reply:
Also I remember mentioning this previously, but this source is only being compiled for Gen 2. Ideally your PR should also decouple these diagnostic classes into a separate source file to be available on both Gen 2 and Gen 3.

Reply:
Thank you for reiterating, we went over a lot of topics that day...

Do you have a suggestion as to where it should live?

Reply:
This has now been split into `system_network_diagnostics.cpp`